JsonDemo/src/main/java/MainInfo.java

```diff
@@ -0,0 +1,52 @@
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class MainInfo {
+    private int temp;
+    private int pressure;
+    private int humidity;
+    @JsonProperty("temp_min")
+    private int tempMin;
+    @JsonProperty("temp_max")
+    private int tempMax;
+
+    public MainInfo(int temp, int pressure, int humidity, int tempMin, int tempMax) {
+        this.temp = temp;
+        this.pressure = pressure;
+        this.humidity = humidity;
+        this.tempMin = tempMin;
+        this.tempMax = tempMax;
+    }
+
+    public MainInfo() {
+
+    }
+
+    public void setTemp(int temp) {
+        this.temp = (int)(temp - 273.15);
```

Comment:
What is this? the whether API allows you to specify the measure units celcius, no need to do custom calculations.